web/modules/custom/dpl_consumers/dpl_consumers.module

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * DPL consumers module.
+ */
+
+/**
+ * Get consumer by Client ID.
+ *
+ * @return string
+ *   The consumer UUID.
+ */
+function dpl_consumers_get_consumer_uuid(string $client_id): string {
+  try {
+    $consumer = \Drupal::entityTypeManager()
+      ->getStorage('consumer')
+      ->loadByProperties(['client_id' => $client_id]);
+
+    if (!empty($consumer)) {
+      $consumer = reset($consumer);
+
+      return $consumer->uuid() ?? 'Consumer UUID not found.';
+    }
+    else {
+      return 'Consumer not found.';
+    }
+  }
+  catch (\Exception $e) {
+    \Drupal::logger('dpl_consumers')->error($e->getMessage());
+    return 'Error fetching consumer.';
+  }
+}
```

Comment:
We should not mix errors and result.
If errors occur we should throw an exception

Reply:
Please take another look, I have exchanged the "faulty" return strings for exceptions instead.